### DISCUSSION TITLE
Maintaining FLAC master and MP3 replica folder structure with Beets

### DISCUSSION THREAD

**Discussion by knuurr** [beetbox/beets#5060] [Q&A] [answered]
Hello,

Currently I'm working on a solution which would allow me to maintain a MP3-transcoded folder mirror of my FLAC-based "primary" folder with music.

The end-goal would ideally meet following criteria:

- original folder with FLACs is NEVER modified in this process 
- any changes to my original library (that is files) such as: adding new files, removing new files, changing metadata, is rellected ONLY one-way; (if I change/add some FLAC file it's reflected in MP3, never other way around)
- during compression, all metadata from input file(s) is kept
- original folder structure is kept

Purpose: all this because I'd like to have a MP3 copy of my music library on my phone locally. FLACs are overkill to keep them - battery drain and storage issues.

My question: how to perform such task using Beets? I tried searching ready to use solutions and ideas on web but I didn't find any ready to use solution.

Is Beets even the right tool for this kind of task? If possible I'd like to avoid pure `ffmpeg`.

**Comment by wisp3rwind** [comment]
Check out https://github.com/geigerzaehler/beets-alternatives

**Reply by knuurr** [reply]
So I run a small test of that on a subset of my libary. Looks fine tbh! I didn't test scenarios like removing from main library and updating, out of band removing of files etc. but in the end I finished with a nice mirror of my main library.

I just wonder if you have any experience if there are no issues with transferring over any tags file might have?

**Reply by wisp3rwind** [reply]
There are some edge cases around removing of files (I don't remember the details now), but updates and resulting file moves in the main beets library should be handled correctly by `beets-alternatives`.

Not sure what exactly you're asking for regarding tags: beets is mostly build around the assumption that it is exclusively in charge of tag writing. Within this assumption, `beets-alternatives` should work fine, it uses the same functions as beets itself to write tags.

**Reply by JOJ0** [reply]
@knuurr yeah beets-alternatives seems to fit your use-case well IIUC, just wanted to mention, there is one edge-case with coverart: If you happen to not save covers as files next to your albums and rely on embeded covers only, that is not supported currently by beets-alternatives. https://github.com/geigerzaehler/beets-alternatives/issues/69

**Reply by knuurr** [reply]
@JOJ0 isn't it the other way? Just ran `beet embedart` on entire library and now my portable music coolection gained cover arts too.

**Reply by JOJ0** [reply]
> @JOJ0 isn't it the other way? Just ran `beet embedart` on entire library and now my portable music coolection gained cover arts too.

It works as long as you have stored a cover.png/jpg inside each album folder in your source/flac structure.

I suppose you do store cover images, that correct?

**Reply by wisp3rwind** [reply, marked as answer]
`beets-alternatives` can definitely embed art that exists as a separate file along the tracks in the converted files. If the art exists only as embedded cover in the source tracks, I _think_ it depends on the `convert` command: `beets-alternatives` will not move over the art itself, but if the conversion command (e.g. `ffmpeg`) does so, it persists.

**Comment by RollingStar** [comment]
I don't have time to write a full answer (not sure this is possible with beets currently) but you will probably want https://github.com/x1ppy/freezetag .

(And have you seen Opus? Great for portable listening and negligible battery drain vs MP3 I would think) see https://hydrogenaud.io/index.php/topic,120919.msg997205.html#msg997205
